python/llm/src/bigdl/llm/transformers/utils.py

```diff
@@ -182,5 +182,5 @@ def quantize_kv_cache(linear: nn.Module, x: torch.Tensor) -> bool:
     if os.environ.get("BIGDL_QUANTIZE_KV_CACHE", None) is not None:
         return os.environ["BIGDL_QUANTIZE_KV_CACHE"] == 1
     else:
-        return x.device.type == 'xpu' and linear.qtype != ggml_tensor_qtype["fp16"] and \
-            linear.qtype != ggml_tensor_qtype["bf16"]
+        return x.device.type == 'xpu' and hasattr(linear, "qtype") and \
+            linear.qtype != ggml_tensor_qtype["fp16"] and linear.qtype != ggml_tensor_qtype["bf16"]
```

Comment:
Does it work? If there are unconverted layers, we will quantize for some layers and not quantize for some other layers.

I think we should record the a static model `qtype` value, instead of checking each layer in a dynamic fashion.

Reply:
we can use a model level `qtype` to avoid check `qtype` dynamically, but we still need to check if is running on XPU dynamically. I think only checking XPU or checking both has no much difference ?